From: Ilya Dryomov <idryomov@gmail.com>
Date: Tue, 27 Oct 2015 16:42:49 +0100
Subject: [PATCH] libceph: drop authorizer check from cephx msg signing
 routines
Git-commit: 4199b8eec36405822619d4176bddfacf7b47eb44
Patch-mainline: v4.4
References: Fate#318586


I don't see a way for auth->authorizer to be NULL in
ceph_x_sign_message() or ceph_x_check_message_signature().

Signed-off-by: Ilya Dryomov <idryomov@gmail.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 net/ceph/auth_x.c |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

--- a/net/ceph/auth_x.c
+++ b/net/ceph/auth_x.c
@@ -697,8 +697,7 @@ static int ceph_x_sign_message(struct ce
 			       struct ceph_msg *msg)
 {
 	int ret;
-	if (!auth->authorizer)
-		return 0;
+
 	ret = calcu_signature((struct ceph_x_authorizer *)auth->authorizer,
 			      msg, &msg->footer.sig);
 	if (ret < 0)
@@ -713,8 +712,6 @@ static int ceph_x_check_message_signatur
 	__le64 sig_check;
 	int ret;
 
-	if (!auth->authorizer)
-		return 0;
 	ret = calcu_signature((struct ceph_x_authorizer *)auth->authorizer,
 			      msg, &sig_check);
 	if (ret < 0)
